GCC: Missing return value check on `fread` in tflite bindings

`bindings/tflite/model.c`: `(void)fread(model->owned_model_data, 1, file_size, file);`

On GCC 7 on Jetson:

```
/home/stella/src/iree/bindings/tflite/include/tensorflow/lite/c/c_api.h: In function ‘TfLiteModelCreateFromFile’:
/home/stella/src/iree/bindings/tflite/model.c:170:3: warning: ignoring return value of ‘fread’, declared with attribute warn_unused_result [-Wunused-result]
   (void)fread(model->owned_model_data, 1, file_size, file);
```

Warning looks legitimate: This should be checking return status.

--- a/bindings/tflite/model.c
+++ b/bindings/tflite/model.c
@@ -159,8 +159,13 @@ TFL_CAPI_EXPORT extern TfLiteModel* TfLiteModelCreateFromFile(
   iree_atomic_ref_count_init(&model->ref_count);
   model->allocator = allocator;
   model->owned_model_data = (uint8_t*)model + file_size;
-  (void)fread(model->owned_model_data, 1, file_size, file);
+  int ret = fread(model->owned_model_data, 1, file_size, file);
   fclose(file);
+  if (ret != file_size) {
+    IREE_TRACE_MESSAGE(ERROR, "failed model+data read");
+    IREE_TRACE_ZONE_END(z0);
+    return NULL;
+  }
 
   status = _TfLiteModelInitializeModule(model->owned_model_data, file_size,
                                         allocator, model);

